### PR TITLE
Use collection expressions in IDE code.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/SignatureHelp/Presentation/Signature.cs [hunk 1]
```diff
@@ -216,8 +216,10 @@ private static IList<TaggedText> AddOptionalBrackets(bool isOptional, IList<Tagg
         {
             if (isOptional)
             {
-                var result = new List<TaggedText>();
-                result.Add(new TaggedText(TextTags.Punctuation, "["));
+                var result = new List<TaggedText>
```
Comment (ToddGrun) [inline src/EditorFeatures/Core.Wpf/SignatureHelp/Presentation/Signature.cs:219]: maybe

                List<TaggedText> result =
                [
                    new TaggedText(TextTags.Punctuation, "["),
                    .. list,
                    new TaggedText(TextTags.Punctuation, "]"),
                ];

File: src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/SyncNamespaceTests_MoveFile.cs [hunk 2]
```diff
@@ -185,9 +185,11 @@ public async Task MoveFile_MultipleAction1()
             var defaultNamespace = "A";
             var declaredNamespace = "A.B.C.D.E";
 
-            var expectedFolders = new List<string[]>();
-            expectedFolders.Add(["B", "C", "D", "E"]);
-            expectedFolders.Add(["B.C", "D", "E"]);
+            var expectedFolders = new List<string[]>
```
Comment (ToddGrun) [inline src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/SyncNamespaceTests_MoveFile.cs:188]: here too?

            List<string[]> expectedFolders = [
                ["B", "C", "D", "E"],
                ["B.C", "D", "E"]
            ];

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SuggestionModeCompletionProviderTests.cs [hunk 3]
```diff
@@ -1505,10 +1505,12 @@ private async Task VerifyWorkerAsync(string markup, bool isBuilder)
 
         private async Task CheckResultsAsync(Document document, int position, bool isBuilder)
         {
-            var triggerInfos = new List<CompletionTrigger>();
-            triggerInfos.Add(CompletionTrigger.CreateInsertionTrigger('a'));
-            triggerInfos.Add(CompletionTrigger.Invoke);
-            triggerInfos.Add(CompletionTrigger.CreateDeletionTrigger('z'));
+            var triggerInfos = new List<CompletionTrigger>
```
Comment (ToddGrun) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SuggestionModeCompletionProviderTests.cs:1508]: >var triggerInfos = new List<CompletionTrigger> [](http://example.com/codeflow?start=12&length=46)

another

File: src/EditorFeatures/Core.Wpf/SignatureHelp/Presentation/Signature.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core.Wpf/SignatureHelp/Presentation/Signature.cs:219]: talked offline.  the nalayzer/fixer here does not want to remove var.  in c# 13 we're looking at expanding collection exprs to work here with `var` as well, so when that comes around, we'll likely want to update this to just work.

Comment (ToddGrun) [reply src/EditorFeatures/Core.Wpf/SignatureHelp/Presentation/Signature.cs:219]: Talked offline, and sounds like the analyzer purposefully doesn't change var semantics. Cyrus indicated there might be a C# 13 feature which magically makes this work with var.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**